src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -254,4 +233,39 @@ public void addNumericField(FieldInfo field, DocValuesProducer valuesProducer) t
     public void close() throws IOException {
         delegatee.close();
     }
+
+    @FunctionalInterface
+    private interface NativeIndexCreator {
+        void createIndex() throws IOException;
+    }
+
+    private void writeFooter(String indexPath, String engineFileName) throws IOException {
+        // Opens the engine file that was created and appends a footer to it. The footer consists of
+        // 1. A Footer magic number (int - 4 bytes)
+        // 2. A checksum algorithm id (int - 4 bytes)
+        // 3. A checksum (long - bytes)
+        // The checksum is computed on all the bytes written to the file up to that point.
+        // Logic where footer is written in Lucene can be found here:
+        // https://github.com/apache/lucene/blob/branch_9_0/lucene/core/src/java/org/apache/lucene/codecs/CodecUtil.java#L390-L412
+        OutputStream os = Files.newOutputStream(Paths.get(indexPath), StandardOpenOption.APPEND);
+        ByteBuffer byteBuffer = ByteBuffer.allocate(8).order(ByteOrder.BIG_ENDIAN);
+        byteBuffer.putInt(FOOTER_MAGIC);
+        byteBuffer.putInt(0);
+        os.write(byteBuffer.array());
+        os.flush();
+
+        ChecksumIndexInput checksumIndexInput = state.directory.openChecksumInput(engineFileName, state.context);
+        checksumIndexInput.seek(checksumIndexInput.length());
+        long value = checksumIndexInput.getChecksum();
+        checksumIndexInput.close();
+
+        if ((value & 0xFFFFFFFF00000000L) != 0) {
```

Comment:
nit: can we use a constant here? also it would make code more readable if condition will be in a separate method with name that describes the idea of the check, e.g. invalidChecksum()